src/core.js

```diff
@@ -45,20 +54,56 @@ class Core extends Model {
 			this.theme = new InfoboxThemeData();
 		}
 
-		// store config for persistence method
-		this.persistOptions = params.persistOptions;
+		let len = routines.length;
+
+		if (len) {
+
+			let bootstrappedRoutines = [];
+
+			for (let i = 0; i < len; i++) {
+				let adapter = routines[i];
+				bootstrappedRoutines.push(Object.assign({}, Serializers[adapter.name], adapter));
```

Comment:
`registerAdapter` registers the adapter. The routines property accepts a different format of adapters to run. They have different purposes.

```typescript
interface Routine {
  name: string;
  persistOptions: any;
}

interface InfoboxTemplateBuilder {
  routines: Routine[];
  static registerAdapter: (adapter: SerializerAdapter): void;
}

interface SerializerAdapter {
  attributes: {
    name: string;
    version: string;
  };
  serialize: (data: InfoboxData, theme: InfoboxThemeData) => string;
  deserialize: (doc: string) => {data: InfoboxData, theme: InfoboxThemeData};
  persist: (data: any, options: any): Promise
}